qutebrowser/browser/commands.py

```diff
@@ -1307,6 +1307,32 @@ def on_editing_finished(self, elem, text):
         except webelem.IsNullError:
             raise cmdexc.CommandError("Element vanished while editing!")
 
+    @cmdutils.register(instance='command-dispatcher',
+                       modes=[KeyMode.insert], hide=True, scope='window')
+    def paste_primary(self):
+        """Paste the primary selection at cursor position into the curently
+        selected form field.
+        """
+        frame = self._current_widget().page().currentFrame()
+        try:
+            elem = webelem.focus_elem(frame)
+        except webelem.IsNullError:
+            raise cmdexc.CommandError("No element focused!")
+        if not elem.is_editable(strict=True):
+            raise cmdexc.CommandError("Focused element is not editable!")
+
+        clipboard = QApplication.clipboard()
+        if clipboard.supportsSelection():
+            sel = clipboard.text(QClipboard.Selection)
+            log.misc.debug("Pasting primary selection into element {} "
+                "(selection is '{}')".format(elem.debug_text(), sel))
+            elem.evaluateJavaScript("""
+                var sel = '{}';
+                var event = document.createEvent('TextEvent');
+                event.initTextEvent('textInput', true, true, null, sel);
+                this.dispatchEvent(event);
+                """.format(sel))
```

Comment:
please dedent this line by one (so only the JS code is indented), and use `webelem.javascript_escape(sel)` so a text containing e.g. `'` works as well.